WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -351,6 +334,58 @@ extension MainTabBarController {
     }
 }
 
+// MARK: - Site ID observation for updating tab view controllers
+//
+private extension MainTabBarController {
+    func observeSiteIDForViewControllers() {
+        cancellableSiteID = stores.siteID.subscribe { [weak self] siteID in
+            guard let self = self else {
+                return
+            }
+            self.updateViewControllers(siteID: siteID)
+        }
+    }
+
+    func updateViewControllers(siteID: Int64?) {
+        guard let siteID = siteID else {
+            viewControllers = []
+            reviewsCoordinator = nil
+            return
+        }
```

Comment:
This PR only has the Reviews tab associated with a site ID, and the other tabs still rely on listening to the site change notification so I keep them around. After we finalize the approach for the Reviews tab, I'll do the same for other tab view controllers so that all of them will be set to `nil` here.